sklearn/svm/_classes.py

```diff
@@ -313,6 +314,25 @@ def fit(self, X, y, sample_weight=None):
         )
         check_classification_targets(y)
         self.classes_ = np.unique(y)
+        if self.kernel == "linear":
+            if self.gamma != "scale":
+                warnings.warn(
+                    "The 'gamma' parameter has been set but is not relevant "
+                    "for the 'linear' kernel. It will be ignored.",
+                    UserWarning,
```

Comment:
These need to be a `FutureWarning` since we're deprecating the usage.

The message should also be more clear that the usage will be an error in 2 release cycles.

You can have a look at our deprecation docs: https://scikit-learn.org/dev/developers/contributing.html#deprecation